add folia support

WhyZerVellasskx opened the pull request:
Hello, I noticed that you don't have folia support, but I need it and I think many users will need it too, so I implemented it.

WhyZerVellasskx committed:
add folia support

WhyZerVellasskx committed:
add folia support

maximjsx commented:
> Hello, I noticed that you don't have folia support, but I need it and I think many users will need it too, so I implemented it.

Did you test it?
Can you also take a look at the merge conflict :D?

WhyZerVellasskx commented:
And now I'll look at the conflicts, I think I've moved on.

@Accessors(chain = true)
protected ItemDisplayMeta.DisplayType displayType = ItemDisplayMeta.DisplayType.FIXED;

@Accessors(chain = true)
protected boolean onFire = false;

@Accessors(chain = true)
protected ItemStack item = new ItemStack.Builder().type(ItemTypes.DIAMOND_BLOCK).build();

@Accessors(chain = true)
protected boolean glowing = false;

@Accessors(chain = true)
protected int glowColor = Color.YELLOW.getRGB();


all setter and getter annotations are on top of the class

WhyZerVellasskx commented:
I'm not sure about testing, I'm just not using this library for plugin development right now. My friend who wrote to you about issue BlackBaroness also uses it. I asked him to test how he would give an answer, I'll tell you, but in fact everything is correct here and nothing should break + I ran the plugin on both versions.

maximjsx commented:
Uhm  ` Cannot invoke "com.tcoded.folialib.FoliaLib.isFolia()" because "com.maximde.hologramlib.utils.BukkitTasks.foliaLib" is null`
you forgot to init folia lib

maximjsx commented:
![image](https://github.com/user-attachments/assets/50bbd8ff-4f49-46b7-8014-4c69e7efdb56)
that should fix it I guess

WhyZerVellasskx commented:
so it worked and there is no bug anymore?

чт, 26 дек. 2024 г. в 00:56, Maxim ***@***.***>:

> Uhm Cannot invoke "com.tcoded.folialib.FoliaLib.isFolia()" because
> "com.maximde.hologramlib.utils.BukkitTasks.foliaLib" is null
> you forgot to init folia lib
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/max1mde/HologramLib/pull/12#issuecomment-2562007699>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3XJWINCFUN3RPMUV4L5MRT2HMSXVAVCNFSM6AAAAABUGJMKPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRSGAYDONRZHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
